iModelCore/ECDb/ECDb/ConcurrentQueryManagerImpl.cpp

```diff
@@ -1171,28 +1188,47 @@ void QueryHelper::Execute(QueryAdaptorCache& adaptorCache, RunnableRequestBase&
         std::string sql = QueryHelper::FormatQuery(request.GetQuery().c_str());
         ECSqlStatus status;
         std::string err;
-        auto adaptor = adaptorCache.TryGet(sql.c_str(), request.UsePrimaryConnection(), request.GetSuppressLogErrors(), status, err);
+
+        // Cache experimental features status to be reset after concurrent query is done
+        const auto experimentalFeaturesStatusCached = adaptorCache.GetConnection().GetPrimaryDb().GetECSqlConfig().GetExperimentalFeaturesEnabled();
+        auto updateExperimentalFeatures = [&adaptorCache] (const bool experimentalFeaturesStatus) { adaptorCache.GetConnection().GetPrimaryDb().GetECSqlConfig().SetExperimentalFeaturesEnabled(experimentalFeaturesStatus); };
+
+        // Enable/Disable experimental features just for this query
+        const auto experimentalFeaturesInRequest = runnableRequest.AreExperimentalFeaturesEnabledForRequest();

```

Comment:
This is a bit nit-picky, but I believe this would be cleaner if you called the function restorePreviousExperimentalFlag() and the variable previousExperimentalFeaturesFlag.
Better name experimentalFeaturesInRequest into experimentalFeaturesEnabledForRequest (the other name may suggest they are actually being used instead of being enabled)